SampleProject/VOKCoreDataManagerTests/VOKThreadingTests.m

```diff
@@ -0,0 +1,37 @@
+//
+//  VOKThreadingTests.m
+//  Vokoder Sample Project
+//
+//  Created by Brock Boland on 2/26/16.
+//  Copyright © 2016 Vokal. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import <VOKCoreDataManager.h>
+
+@interface VOKThreadingTests : XCTestCase
+
+@end
+
+@implementation VOKThreadingTests
+
+- (void)testSetupOnBackgroundThreadFails
+{
+    XCTestExpectation *willFinish = [self expectationWithDescription:@"Background thread setup will finish"];
+
+    // Setup on the main thread should work
+    [[VOKCoreDataManager sharedInstance] resetCoreData];
+    XCTAssertNoThrow([[VOKCoreDataManager sharedInstance] setResource:@"VOKCoreDataModel" database:nil]);
+
+    NSOperationQueue *backgroundQueue = [[NSOperationQueue alloc] init];
+    backgroundQueue.qualityOfService = NSOperationQualityOfServiceBackground;
+    [backgroundQueue addOperationWithBlock:^{
+        [[VOKCoreDataManager sharedInstance] resetCoreData];
+        // Setup on a background thread should fail
```

Comment:
The test fails because of the `XCTAssertThrows`, but to your real question: if I remove the assertions in VOKCoreDataManager (including the existing one), and then use this background thread to setup the MOC and create an object, CoreData seems to be OK. Same is true if I create the MOC on the main thread and then create an object in the background thread. I think problems will only surface later on in more complicated, real-world situations that I'm not sure how to test well, which is why I added this better-than-nothing-but-not-great test.